plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Definition.hs

```diff
@@ -220,6 +219,86 @@ test_ExpensivePlus =
             traverse (try . evaluate) $ typecheckEvaluateCekNoEmit defaultCekParametersExt term
         typeErrOrEvalExcOrRes @?= Right (Right EvaluationFailure)
 
+-- | Test that @Null@, @Head@ and @Tail@ are enough to get pattern matching on built-in lists.
+test_BuiltinList :: TestTree
+test_BuiltinList =
```

Comment:
I fucked up a merge in the grand built-in lists and tuples PR and we lost all its tests. This PR fixes that.